cabal-env/src/CabalEnv/Main.hs

```diff
@@ -337,7 +341,33 @@ generateEnvironment tracer Opts {..} env ghcInfo plan planBS = do
     putDebug tracer "writing environment file"
     when optVerbose $ outputErr tracer newEnvFileContents
 
-    writeByteString (ghcEnvDir ghcInfo </> fromUnrootedFilePath optEnvName) (toUTF8BS newEnvFileContents)
+    writeByteString envPath (toUTF8BS newEnvFileContents)
+
+-- | Obtain the path to the environment file the user is referring to.
+--
+-- Mirroring the behaviour of ghc and cabal install --lib, a path-like string
+-- is interpreted literally, and a non-path-like one is assumed to be in
+-- the global environment directory.
+--
+-- "foo" -> $ghcEnvDir/foo
+-- "./" -> $pwd/.ghc.environment.$ghcPlarform-$ghcVersion
+-- "./foo" where foo is not a directory -> $pwd/foo
+getEnvPath :: Opts -> GhcInfo -> Peu r (Path Absolute)
+getEnvPath Opts { optEnvName } ghcInfo =
+    if FP.takeBaseName optEnvName == optEnvName
```

Comment:
this is how it's done in cabal, but I'm not sure it's the best way to do it

Reply:
Yes, I don't like this either. I think picking some format (e.g. alphanumeric + `-_`) to be environment names, and everything else interpreted as paths would be better.

I'm not sure. I'll think some more and do the changes myself. Bear with me.

Reply:
Is there anything I can do to help with this?